stdlib/REPL/src/REPL.jl

```diff
@@ -1225,18 +1228,20 @@ function setup_interface(
         end,
         ']' => function (s::MIState,o...)
             if isempty(s) || position(LineEdit.buffer(s)) == 0
-                pkgid = Base.PkgId(Base.UUID("44cfe95a-1eb2-52ea-b672-e2afdf69b78f"), "Pkg")
-                REPLExt = Base.require_stdlib(pkgid, "REPLExt")
-                pkg_mode = nothing
-                if REPLExt isa Module && isdefined(REPLExt, :PkgCompletionProvider)
-                    for mode in repl.interface.modes
-                        if mode isa LineEdit.Prompt && mode.complete isa REPLExt.PkgCompletionProvider
-                            pkg_mode = mode
-                            break
+                global pkg_mode
+                if pkg_mode === nothing
```

Comment:
This may be a really stupid question, as I know nothing about this code, but: is it OK that `pkg_mode` is never reset to `nothing`? I.e. this is about getting a pointer to a specific `LineEdi.Prompt` instance, and once you have it, you never have to look for it again -- correct?

Reply:
correct